cmake/modules/FindGINKGO.cmake

```diff
@@ -64,9 +80,9 @@ DEAL_II_FIND_PATH(GINKGO_INCLUDE_DIR ginkgo/ginkgo.hpp
 
 DEAL_II_PACKAGE_HANDLE(GINKGO
   LIBRARIES
-    REQUIRED GINKGO_LIBRARY GINKGO_REFERENCE_LIBRARY GINKGO_OMP_LIBRARY GINKGO_CUDA_LIBRARY
+    REQUIRED GINKGO_LIBRARY GINKGO_REFERENCE_LIBRARY GINKGO_OMP_LIBRARY GINKGO_CUDA_LIBRARY GINKGO_HIP_LIBRARY GINKGO_DPCPP_LIBRARY
```

Comment:
Does gingko *always* build for HIP and DPCPP? I'm asking because if you're on a CPU-only platform, will this actually work?

Reply:
No, the different libraries (`ginkgo_cuda`, `ginkgo_hip`, ... ) are always *created*. When on a platform that does not support any of the specific compilers/devices needed for the modules then these libraries contain dummy implementations. 

It is safe to assume that these libraries will always be available (with the appropriate version of Ginkgo) and for example if you are on a CPU-only system and try to create a `HipExecutor` to run on AMD GPU, then a runtime exception is thrown.